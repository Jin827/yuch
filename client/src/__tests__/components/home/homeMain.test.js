import React from 'react';
import { render, cleanup } from '../../setupTests';
import HomeMain from '../../../components/home/homeMain';

afterEach(cleanup);

const setUp = () => render(<HomeMain />);

test('<HomeMain /> renders properly', () => {
  const { container } = setUp();
  expect(container.firstChild).toMatchSnapshot();
});

test('images and links have proper path', () => {
  const { getByTestId, queryByTestId } = setUp();

  // image
  expect(getByTestId('img-delivery').getAttribute('src')).toBe('delivery.jpg');
  expect(queryByTestId('img-localfood').getAttribute('data-src')).toBe(
    'localfood.jpg',
  );
  expect(queryByTestId('img-sesame').getAttribute('data-src')).toBe(
    'sesame.jpg',
  );
  expect(queryByTestId('img-women').getAttribute('data-src')).toBe('women.jpg');

  // link
  expect(getByTestId('sesame-link').getAttribute('href')).toBe(
    'http://www.xn--0z2bz4lsqjfrj.com/',
  );
});
