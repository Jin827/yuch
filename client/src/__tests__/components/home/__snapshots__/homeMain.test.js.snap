// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomeMain /> renders properly 1`] = `
<div
  class="home-main--container"
  id="home-main"
>
  <div
    class="home-main--box"
  >
    <img
      alt="신선한 음식 배달"
      class="home-main--img"
      data-testid="img-delivery"
      src="delivery.jpg"
    />
    <div
      class="home-main--text"
    >
      <p
        class="f-mini lh-3"
      >
        OUR RESTAURANT
      </p>
      <h3
        class="c-text1"
      >
        당신의 위한 
        유청
         서비스
      </h3>
      <div
        class="mb3"
      >
        <h4
          class="b"
        >
          위탁급식
        </h4>
        <p>
          기업계, 각종 행사 모임
          <br />
          연중 중식, 석식, 야식 배달
        </p>
      </div>
      <div>
        <h4
          class="b"
        >
          식당
        </h4>
        <p>
          중식, 석식 뷔폐 서비스
        </p>
      </div>
    </div>
  </div>
  <div
    class="home-main--box reverse"
  >
    <img
      alt="지역 농산품"
      class="lazy, home-main--img"
      data-src="localfood.jpg"
      data-testid="img-localfood"
    />
    <div
      class="home-main--text"
    >
      <p
        class="f-mini lh-3"
      >
        HEALTHY FOOD
      </p>
      <h3
        class="c-text1"
      >
        신선한 지역 농산물
      </h3>
      <p>
        지역 농산물 생산 · 가공 · 소비 활성화를 위하여, 경주 지역민들의 밭에서 공수해온 신선한 농산물들을 주재료로 사용합니다.
      </p>
    </div>
  </div>
  <div
    class="home-main--box"
  >
    <img
      alt="참기름"
      class="lazy, home-main--img"
      data-src="sesame.jpg"
      data-testid="img-sesame"
    />
    <div
      class="home-main--text"
    >
      <p
        class="f-mini lh-3"
      >
        OUR COMMUNITY
      </p>
      <h3
        class="c-text1"
      >
        경주시 지원사업 장려
        <br />
        100% 통참깨 참기름 사용
      </h3>
      <p>
        경주 시니어 클럽에서 주관하는 노인 일자리 창출 사업을 장려하기 위해 
        <a
          class="sesame-link td-none b"
          data-testid="sesame-link"
          href="http://www.xn--0z2bz4lsqjfrj.com/"
          rel="noopener noreferrer"
          target="_blank"
        >
          착한손맛
        </a>
        의 참기름을 사용합니다.
      </p>
    </div>
  </div>
  <div
    class="home-main--box reverse"
  >
    <img
      alt="여성을 위한 기업"
      class="lazy, home-main--img"
      data-src="women.jpg"
      data-testid="img-women"
    />
    <div
      class="home-main--text"
    >
      <p
        class="f-mini lh-3"
      >
        CAREER WOMEN
      </p>
      <h3
        class="c-text1"
      >
        여성이 일하기 좋은 기업
      </h3>
      <p>
        2016년 여성이 일하기 좋은 기업으로 선정.
        <br />
        유청
        은 나이와 국적에 제한을 두지 않고 채용합니다.
      </p>
    </div>
  </div>
</div>
`;
